test/integration/master/synthetic_master_test.go

```diff
@@ -224,6 +225,101 @@ func TestStatus(t *testing.T) {
 		}
 	}
 }
+func constructBodyByManyLabels(val string, size int) string {
+
+	var temp = `{
+		"apiVersion": "apps/v1beta1",
+		"kind": "Deployment",
+		"metadata": {
+		   "name": "test-deployment2",
+		   "namespace": "default",
+		   "labels": {`
+
+	for i := 0; i < size; i++ {
+		temp += "\"test" + strconv.Itoa(i) + "\" : \"" + val + "\" ,"
+	}
+
+	temp += `"test100000" :"a" 
+			}
+			},
+			"spec": {
+			"replicas": 1,
+			"template": {
+			"metadata": {
+				"labels": {
+				"foo": "bar"
+				}
+			},
+			"spec": {
+				"containers": [{
+				"name": "foo",
+				"image": "foo"
+				}]
+			}
+			}
+			}
+			}`
+
+	return temp
+}
+
+func TestObjectSizeResponses(t *testing.T) {
+
+	_, s, closeFn := framework.RunAMaster(nil)
+	defer closeFn()
+	transport := http.DefaultTransport
+
+	var expectedMsgFor1MB = `etcdserver: request is too large`
+	var expectedMsgFor2MB = `rpc error: code = ResourceExhausted desc = grpc: trying to send message larger than max`
+
+	deployment1 := constructBodyByManyLabels("a", 100000)                    // >1 MB file
+	deployment2 := constructBodyByManyLabels("aaaaaaaaaaaaaaaaaaaa", 100000) // >2 MB file
+
+	//deployment3 := constructBodyByManyAnnotations("a", 100000)
+
+	requests := []struct {
+		size            string
+		verb            string
+		URL             string
+		body            string
+		expectedMessage string
+	}{
+		{"1 MB", "POST", "/apis/apps/v1beta1/namespaces/" + metav1.NamespaceDefault + "/deployments", deployment1, expectedMsgFor1MB},
+		{"2 MB", "POST", "/apis/apps/v1beta1/namespaces/" + metav1.NamespaceDefault + "/deployments", deployment2, expectedMsgFor2MB},
+	}
+
+	for _, r := range requests {
+		bodyBytes := bytes.NewReader([]byte(r.body))
+		req, err := http.NewRequest(r.verb, s.URL+r.URL, bodyBytes)
+		if err != nil {
+			t.Logf("case %v", r)
+			t.Fatalf("unexpected error: %v", err)
+		}
+		func() {
+			resp, err := transport.RoundTrip(req)
+			defer resp.Body.Close()
+			if err != nil {
+				t.Logf("case %v", r)
+				t.Fatalf("unexpected error: %v", err)
```

Comment:
Just put the Logf output into the Fatalf string.

Reply:
Done